src/System.Linq.Parallel/tests/DegreeOfParallelismTests.cs

```diff
@@ -1,140 +1,86 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Xunit;
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Xunit;
 
 namespace Test
 {
     public class DegreeOfParallelismTests
     {
-        [Fact]
-        [OuterLoop]
-        public static void RunDopTests()
+        public static IEnumerable<object[]> DegreeData(object[] counts, object[] degrees)
         {
-            RunDopTest(1, false);
-            RunDopTest(4, false);
-            RunDopTest(512, false);
-            RunDopTest(513, true);
+            foreach (object[] results in UnorderedSources.Ranges(counts.Cast<int>()))
+            {
+                foreach (int degree in degrees.DefaultIfEmpty(results[1]).Cast<int>())
+                {
+                    yield return new object[] { results[0], results[1], degree };
+                }
+            }
         }
 
-        [Fact]
+        [Theory]
+        [MemberData("DegreeData", new[] { 1024 }, new[] { 1, 4, 512 })]
         [OuterLoop]
-        public static void RunDopBarrierTest()
+        public static void DegreeOfParallelism(Labelled<ParallelQuery<int>> labelled, int count, int degree)
         {
-            RunDopBarrierTestCore(1);
-            RunDopBarrierTestCore(4);
-            RunDopBarrierTestCore(32);
-            // These variations take way too long and hit the 100s timeout.
-            //RunDopBarrierTest(64);
-            //RunDopBarrierTest(512);
+            Assert.Equal(Functions.SumRange(0, count), labelled.Item.WithDegreeOfParallelism(degree).Sum());
         }
 
-        [Fact]
+        [Theory]
+        [MemberData("DegreeData", new[] { 1, 4, 32 }, new object[] { })]
         [OuterLoop]
-        public static void RunDopPipeliningTest()
+        public static void DegreeOfParallelism_Barrier(Labelled<ParallelQuery<int>> labelled, int count, int degree)
         {
-            RunDopPipeliningTestCore(1000000, 1);
-            RunDopPipeliningTestCore(1000000, 4);
-            RunDopPipeliningTestCore(1000000, 64);
-            RunDopPipeliningTestCore(1000000, 512);
+            //int minWorker, minAsyncIO;
+            //ThreadPool.GetMinThreads(out minWorker, out minAsyncIO);
+            //ThreadPool.SetMinThreads(degree, minAsyncIO);
```

Comment:
When I last wrote it, `ThreadPool` (or maybe just those methods) wasn't available.  I'll check again.